source/otus-business/src/main/java/br/org/otus/participant/api/ParticipantFacade.java

```diff
@@ -142,7 +146,19 @@ public Boolean editEmail(String participantId, String email) {
       if(!SystemInstallationEmail.isValid(email)) {
         throw new ValidationException(new Throwable("Malformed email"));
       }
-      return participantService.editEmail(participantId, email);
+
+      if(!ObjectId.isValid(participantId)) {
+        throw new ValidationException(new Throwable("ObjectId is not valid"));
+      }
+
+      ObjectId id = new ObjectId(participantId);
+      Boolean result = participantService.editEmail(id, email);
+
+      if(result){
+        activityService.updateEmailByParticipant(participantService.getId(id));
```

Comment:
seria melhor se chamasse o facade de atividade, não o serviço.
participantService.getId(id) não faz sentido. Tá fazendo o get de um id passando um id? Acredito que só get seria suficiente, caso seja um get de participant

Reply:
Em uso ActivityFacade causou error deploy, segue o log:
`2020-07-24T17:33:47.016457003Z 17:33:47,015 ERROR [org.jboss.as.controller.management-operation] (Controller Boot Thread) WFLYCTL0013: Operation ("deploy") failed - address: ([("deployment" => "otus-ear.ear")]) - failure description: {"WFLYCTL0080: Failed services" => {"jboss.deployment.unit.\"otus-ear.ear\".WeldStartService" => "Failed to start service
2020-07-24T17:33:47.016479449Z     Caused by: org.jboss.weld.exceptions.DeploymentException: Exception List with 5 exceptions:
2020-07-24T17:33:47.016483613Z Exception 0 :
2020-07-24T17:33:47.016486612Z org.jboss.weld.exceptions.DeploymentException: WELD-001443: Pseudo scoped bean has circular dependencies. Dependency path: 
2020-07-24T17:33:47.016489639Z   - Managed Bean [class br.org.otus.outcomes.FollowUpFacade] with qualifiers [@Any @Default],
2020-07-24T17:33:47.016492712Z   - [BackedAnnotatedField] @Inject private br.org.otus.outcomes.FollowUpFacade.participantFacade,
2020-07-24T17:33:47.016495724Z   - Managed Bean [class br.org.otus.participant.api.ParticipantFacade] with qualifiers [@Any @Default],
2020-07-24T17:33:47.016498787Z   - [BackedAnnotatedField] @Inject private br.org.otus.participant.api.ParticipantFacade.activityFacade,
2020-07-24T17:33:47.016501751Z   - Managed Bean [class br.org.otus.survey.activity.api.ActivityFacade] with qualifiers [@Any @Default],
2020-07-24T17:33:47.016504783Z   - [BackedAnnotatedField] @Inject private br.org.otus.survey.activity.api.ActivityFacade.followUpFacade,
2020-07-24T17:33:47.016507827Z   - Managed Bean [class br.org.otus.outcomes.FollowUpFacade] with qualifiers [@Any @Default]
2020-07-24T17:33:47.016510827Z 	at org.jboss.weld.bootstrap.Validator.reallyValidatePseudoScopedBean(Validator.java:942)
2020-07-24T17:33:47.016513795Z 	at org.jboss.weld.bootstrap.Validator.validatePseudoScopedInjectionPoint(Validator.java:989)
2020-07-24T17:33:47.016516878Z 	at org.jboss.weld.bootstrap.Validator.reallyValidatePseudoScopedBean(Validator.java:951)
2020-07-24T17:33:47.016520003Z 	at org.jboss.weld.bootstrap.Validator.validatePseudoScopedInjectionPoint(Validator.java:989)
2020-07-24T17:33:47.016523014Z 	at org.jboss.weld.bootstrap.Validator.reallyValidatePseudoScopedBean(Validator.java:951)
2020-07-24T17:33:47.016525970Z 	at org.jboss.weld.bootstrap.Validator.validatePseudoScopedInjectionPoint(Validator.java:989)
2020-07-24T17:33:47.016529056Z 	at org.jboss.weld.bootstrap.Validator.reallyValidatePseudoScopedBean(Validator.java:951)
2020-07-24T17:33:47.016538416Z 	at org.jboss.weld.bootstrap.Validator.validatePseudoScopedBean(Validator.java:928)
2020-07-24T17:33:47.016542894Z 	at org.jboss.weld.bootstrap.Validator.validateGeneralBean(Validator.java:147)
2020-07-24T17:33:47.016545883Z 	at org.jboss.weld.bootstrap.Validator.validateRIBean(Validator.java:164)
2020-07-24T17:33:47.016548789Z 	at org.jboss.weld.bootstrap.Validator.validateBean(Validator.java:526)
2020-07-24T17:33:47.016551719Z 	at org.jboss.weld.bootstrap.ConcurrentValidator$1.doWork(ConcurrentValidator.java:64)
2020-07-24T17:33:47.016554645Z 	at org.jboss.weld.bootstrap.ConcurrentValidator$1.doWork(ConcurrentValidator.java:62)
2020-07-24T17:33:47.016557747Z 	at org.jboss.weld.executor.IterativeWorkerTaskFactory$1.call(IterativeWorkerTaskFactory.java:62)
2020-07-24T17:33:47.016560755Z 	at org.jboss.weld.executor.IterativeWorkerTaskFactory$1.call(IterativeWorkerTaskFactory.java:55)
2020-07-24T17:33:47.016563767Z 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
2020-07-24T17:33:47.016566764Z 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
2020-07-24T17:33:47.016569663Z 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
2020-07-24T17:33:47.016572590Z 	at java.lang.Thread.run(Thread.java:748)
2020-07-24T17:33:47.016575539Z 	at org.jboss.threads.JBossThread.run(JBossThread.java:485)`